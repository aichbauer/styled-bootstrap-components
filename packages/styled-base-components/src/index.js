import { BootstrapBaseCss } from './components';

export { BootstrapBaseCss };
