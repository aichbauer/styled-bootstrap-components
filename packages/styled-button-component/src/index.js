import {
  Button,
  LinkButton,
} from './components/Button';
import { ButtonGroup } from './components/ButtonGroup';
import { ButtonToolbar } from './components/ButtonToolbar';

export {
  Button,
  LinkButton,
  ButtonGroup,
  ButtonToolbar,
};
