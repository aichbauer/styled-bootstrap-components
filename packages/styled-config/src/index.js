import { theme } from './theme';
import {
  border,
  borderRadius,
  boxShadow,
  colors,
  fontFamily,
  fontSize,
  fontWeight,
  height,
  margin,
  padding,
  screenSize,
  width,
} from './utils';

export {
  theme,
  // utils
  border,
  borderRadius,
  boxShadow,
  colors,
  fontFamily,
  fontSize,
  fontWeight,
  height,
  margin,
  padding,
  screenSize,
  width,
};
