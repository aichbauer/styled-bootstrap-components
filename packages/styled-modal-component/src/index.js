import { Modal } from './components/Modal';
import { ModalWrapper } from './components/ModalWrapper';
import { ModalBody } from './components/ModalBody';
import { ModalContent } from './components/ModalContent';
import { ModalDialog } from './components/ModalDialog';
import { ModalFooter } from './components/ModalFooter';
import { ModalHeader } from './components/ModalHeader';
import { ModalTitle } from './components/ModalTitle';
import { ModalBackdrop } from './components/ModalBackdrop';

export {
  Modal,
  ModalWrapper,
  ModalBody,
  ModalContent,
  ModalDialog,
  ModalFooter,
  ModalHeader,
  ModalTitle,
  ModalBackdrop,
};
