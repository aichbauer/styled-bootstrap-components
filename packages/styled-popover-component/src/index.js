import { PopoverWithoutTransition, Popover } from './components/Popover';
import { PopoverArrow } from './components/PopoverArrow';
import { PopoverBody } from './components/PopoverBody';
import { PopoverHeader } from './components/PopoverHeader';

export {
  PopoverWithoutTransition,
  Popover,
  PopoverArrow,
  PopoverBody,
  PopoverHeader,
};
