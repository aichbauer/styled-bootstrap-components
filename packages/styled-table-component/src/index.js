import { Table } from './components/Table';
import { Tr } from './components/Tr';

export {
  Table,
  Tr,
};
