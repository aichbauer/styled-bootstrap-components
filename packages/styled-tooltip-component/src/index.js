import { TooltipWithoutTransition, Tooltip } from './components/Tooltip';
import { TooltipArrow } from './components/TooltipArrow';
import { TooltipInner } from './components/TooltipInner';

export {
  TooltipWithoutTransition,
  Tooltip,
  TooltipArrow,
  TooltipInner,
};
