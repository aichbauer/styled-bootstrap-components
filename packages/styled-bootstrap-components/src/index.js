import { Alert } from 'styled-alert-component';
import { Badge } from 'styled-badge-component';
import {
  A,
  Abbr,
  Address,
  Article,
  Aside,
  B,
  Blockquote,
  Button as BaseButton,
  Caption,
  Code,
  Dd,
  Dfn,
  Dialog,
  Dl,
  Dt,
  Fieldset,
  Figcaption,
  Figure,
  Footer,
  H1,
  H2,
  H3,
  H4,
  H5,
  H6,
  Header,
  Hgroup,
  Hr,
  Img,
  Input,
  Kbd,
  Legend,
  Label,
  Main,
  Nav as Navigation,
  Ol,
  Optgroup,
  Output,
  P,
  Pre,
  Progress,
  Samp,
  Section,
  Select,
  Small,
  Strong,
  Sub,
  Summary,
  Sup,
  Svg,
  Table as BaseTable,
  Textarea,
  Th,
  Ul,
} from 'styled-base-components';
import {
  Breadcrumb,
  BreadcrumbItem,
} from 'styled-breadcrumb-component';
import {
  Button,
  ButtonGroup,
  ButtonToolbar,
} from 'styled-button-component';
import {
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  CardImageHeader,
  CardText,
  CardTitle,
} from 'styled-card-component';
import { screenSize } from 'styled-config';
import { Container } from 'styled-container-component';
import {
  Dropdown,
  DropdownItem,
  DropdownMenu,
  DropdownDivider,
} from 'styled-dropdown-component';
import {
  FormCheck,
  FormCheckInput,
  FormControl,
  FormControlFile,
  FormControlPlainText,
  FormGroup,
  FormInline,
  FormText,
  InputGroup,
  InputGroupAppend,
  InputGroupPrepend,
  InputGroupText,
} from 'styled-form-component';
import {
  Column,
  Row,
} from 'styled-grid-system-component';
import { Jumbotron } from 'styled-jumbotron-component';
import {
  ListGroup,
  ListGroupItem,
} from 'styled-listgroup-component';
import {
  Modal,
  ModalBody,
  ModalContent,
  ModalDialog,
  ModalFooter,
  ModalHeader,
  ModalTitle,
} from 'styled-modal-component';
import {
  Nav,
  NavLink,
} from 'styled-nav-component';
import {
  Navbar,
  NavbarLink,
} from 'styled-navbar-component';
import {
  Popover,
  PopoverArrow,
  PopoverBody,
  PopoverHeader,
} from 'styled-popover-component';
import {
  Table,
  Tr,
} from 'styled-table-component';
import {
  Tooltip,
  TooltipArrow,
  TooltipInner,
} from 'styled-tooltip-component';

export {
  Alert,
  Badge,
  A,
  Abbr,
  Address,
  Article,
  Aside,
  B,
  Blockquote,
  BaseButton,
  Caption,
  Code,
  Dd,
  Dfn,
  Dialog,
  Dl,
  Dt,
  Fieldset,
  Figcaption,
  Figure,
  Footer,
  H1,
  H2,
  H3,
  H4,
  H5,
  H6,
  Header,
  Hgroup,
  Hr,
  Img,
  Input,
  Kbd,
  Legend,
  Label,
  Main,
  Navigation,
  Ol,
  Optgroup,
  Output,
  P,
  Pre,
  Progress,
  Samp,
  Section,
  Select,
  Small,
  Strong,
  Sub,
  Summary,
  Sup,
  Svg,
  BaseTable,
  Textarea,
  Th,
  Ul,
  Breadcrumb,
  BreadcrumbItem,
  Button,
  ButtonGroup,
  ButtonToolbar,
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  CardImageHeader,
  CardText,
  CardTitle,
  screenSize,
  Container,
  Dropdown,
  DropdownItem,
  DropdownMenu,
  DropdownDivider,
  FormCheck,
  FormCheckInput,
  FormControl,
  FormControlFile,
  FormControlPlainText,
  FormGroup,
  FormInline,
  FormText,
  InputGroup,
  InputGroupAppend,
  InputGroupPrepend,
  InputGroupText,
  Column,
  Row,
  Jumbotron,
  ListGroup,
  ListGroupItem,
  Modal,
  ModalBody,
  ModalContent,
  ModalDialog,
  ModalFooter,
  ModalHeader,
  ModalTitle,
  Nav,
  NavLink,
  Navbar,
  NavbarLink,
  Popover,
  PopoverArrow,
  PopoverBody,
  PopoverHeader,
  Table,
  Tr,
  Tooltip,
  TooltipArrow,
  TooltipInner,
};
