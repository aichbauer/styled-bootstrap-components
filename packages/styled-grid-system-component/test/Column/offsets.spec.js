import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { Column } from '../../src';

// TODO: how to check @media queries
test('Styles matches Column with specified xs offset', () => {
  const tree = renderer.create(<Column xsOffset={6} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Styles matches Column with specified sm offset', () => {
  const tree = renderer.create(<Column smOffset={6} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Styles matches Column with specified md offset', () => {
  const tree = renderer.create(<Column mdOffset={6} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Styles matches Column with specified lg offset', () => {
  const tree = renderer.create(<Column lgOffset={6} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Styles matches Column with specified xl offset', () => {
  const tree = renderer.create(<Column xlOffset={6} />).toJSON();
  expect(tree).toMatchSnapshot();
});
