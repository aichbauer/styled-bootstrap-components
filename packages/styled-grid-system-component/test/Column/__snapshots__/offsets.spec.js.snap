// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styles matches Column with specified lg offset 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  padding-right: 15px;
  padding-left: 15px;
  width: 100%;
}

@media (min-width:576px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

@media (min-width:992px) {
  .c0 {
    margin-left: 50%;
  }
}

<div
  className="c0"
/>
`;

exports[`Styles matches Column with specified md offset 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  padding-right: 15px;
  padding-left: 15px;
  width: 100%;
}

@media (min-width:576px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

@media (min-width:768px) {
  .c0 {
    margin-left: 50%;
  }
}

<div
  className="c0"
/>
`;

exports[`Styles matches Column with specified sm offset 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  padding-right: 15px;
  padding-left: 15px;
  width: 100%;
}

@media (min-width:576px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    margin-left: 50%;
  }
}

<div
  className="c0"
/>
`;

exports[`Styles matches Column with specified xl offset 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  padding-right: 15px;
  padding-left: 15px;
  width: 100%;
}

@media (min-width:576px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

@media (min-width:1200px) {
  .c0 {
    margin-left: 50%;
  }
}

<div
  className="c0"
/>
`;

exports[`Styles matches Column with specified xs offset 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  padding-right: 15px;
  padding-left: 15px;
  width: 100%;
  margin-left: 50%;
}

@media (min-width:576px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

<div
  className="c0"
/>
`;
